jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -228,9 +242,27 @@ public Resource getQuickStartWebXml(WebAppContext context) throws Exception
         Resource webInf = context.getWebInf();
         if (webInf == null || !webInf.exists())
         {
-            File tmp = new File(context.getBaseResource().getFile(), "WEB-INF");
-            tmp.mkdirs();
-            webInf = context.getWebInf();
+            switch (getModeForContext(context))
```

Comment:
There's only 1 mode where a `quickstart-web.xml` will be generated, and that's `GENERATE`.

Reply:
Fixed